myapp/spec/models/users_spec.rb

```diff
@@ -0,0 +1,65 @@
+require 'rails_helper'
+
+describe User, type: :model do
+  describe '#validation' do
+    describe 'email' do
+      context 'email設定' do
+        subject(:user) { FactoryBot.build(:user, email: 'test@example.com') }
+
+        it { is_expected.to be_valid }
+      end
+
+      context 'email未設定' do
+        subject(:user) { FactoryBot.build(:user, email: '') }
+
+        it { is_expected.to be_invalid }
+      end
+
+      context 'email重複登録' do
+        let!(:user) { FactoryBot.create(:user, email: 'test@example.com') }
+
+        it { expect { FactoryBot.create(:user, email: 'test@example.com') }.to raise_error(ActiveRecord::RecordInvalid) }
```

Comment:
これでもいいですが、他と同じようにcreateではなくbuildしたオブジェクトにバリデーションを実行するで良いでのは？

Reply:
私の認識ですが、buildはオブジェクトを作成し、DBに登録する前の状態だと思っています。
User1をDBに登録→User2をDBに登録→emailが重複し、エラーが発生
の流れでないとテストできないかなと思いcreateにしました。
→実際最初buildにしていたのですが、エラーが発生せず正常にDBに登録されていました。

こちらを回避するよい法法がありましたらお教えいただきたいです。

Reply:
もしかしたら、subjectとlet!でuserという変数名が被っているのが原因かもしれません。
このテストではUser1をlet!で作成していますが、beforeでも問題なく動くはずです。
その上で、itは他のテストと同様の書き方ができると思います。
```suggestion
        it { is_expected.to be_invalid }
```

Reply:
buildへ修正しました。
ただ、before内で事前にデータを登録するところはcreateで実際にDBに保存する必要があったためそちらはcreateにしています。